src/send/mod.rs

```diff
@@ -49,10 +49,10 @@ pub fn get_writer<'a>(
     level: Level,
 ) -> Result<Box<dyn io::Write + Send + 'a>, Error> {
     match format {
-        compression::Format::Gzip => compression::new_gz_encoder(out_stream, level),
-        compression::Format::Bzip => compression::new_bz2_encoder(out_stream, level),
-        compression::Format::Lzma => compression::new_lzma_encoder(out_stream, level),
-        compression::Format::Zstd => compression::new_zstd_encoder(out_stream, level),
+        compression::Format::Gzip => compression::gz::encoder(out_stream, level),
```

Comment:
now each format is a submodule instead of a function, and all submodules have an `encoder` and a `decoder` function. These are all private, so it doesn't break the public API